api/lds.proto

```diff
@@ -205,3 +212,16 @@ message FilterChain {
   // [#not-implemented-hide:] See base.TransportSocket description.
   TransportSocket transport_socket = 6;
 }
+
+message ListenerFilter {
+  // The name of the filter to instantiate. The name must match a supported
+  // filter. The built-in filters are:
+  //
+  // [#comment:TODO(mattklein123): Auto generate the following list]
+  // * :ref:`envoy.listener.original_dst <config_listener_filters_original_dst>`
```

Comment:
Should we deprecate the existing original_dst field? I think we should. This means:
1) Add deprecated annotation to proto field.
2) Adding doc note about deprecation and pointing people to here.
3) Updating DEPRECATED.md in the main repo.

Reply:
One issue about this I have been thinking about is that doing a search for a forward-to listener is kind of expensive when there are many listeners, and likely not needed when v2 filter chain matching is implemented. I'd like to deprecate this search together with this flag, if possible. If not possible, I'd like to keep this flag and perform the search for another listener only if this flag is set. I.e., if original destination feature is configured as a listener filter, the search for another listener would not be done. Thoughts?

Reply:
> I'd like to deprecate this search together with this flag, if possible.

Agreed. I guess if you don't want to deprecate until the full filter chain matching is implemented, that's fine, but I would put a TODO/comment somewhere. Or we can just mark up the deprecation now in this PR.

Reply:
IMO marking for deprecation now is fine, anyone depending on this can still use v1 API regardless when filter chain matching is implemented. I'll mark this as TODO on the Envoy-side PR.

Reply:
SGTM

Reply:
Do you have an example of a deprecated annotation to a proto field?

Reply:
I guess you meant deprecated annotation to the doc comment above the field definition, but then points 1. & 2. would be the same point?

Reply:
https://github.com/envoyproxy/data-plane-api/blob/master/api/bootstrap.proto#L222

Reply:
No, I mean annotation per previous comment, then in the doc, then in DEPRECATED.md

Reply:
Do you think this would work as the doc annotation:
```
This field is deprecated. Use :ref:`an original_dst <config_listener_filters_original_dst>`
:ref:`listener filter <envoy_api_field_Listener.filter_filters>` instead. Note that hand off
to another listener is *NOT* performed without this flag. :ref:`FilterChainMatch
<envoy_api_msg_FilterChainMatch>` may be used to select a filter chain based on the restored destination address instead.
```

Reply:
I would make it more clear that in the current code, handoff is not performed without this flag, and in the future when full filter chain matching is implemented it will be completely replaced with filter chain matching. Basically, it is partially deprecated. We can fully deprecate it when we do the full filter chain matching PR. Does that make sense?

Reply:
Not sure of the syntax to use for a boolean, would this work:
```
  google.protobuf.BoolValue use_original_dst = 4 [(validate.rules) deprecated = true];
```

Reply:
Your comment above for additional clarity makes perfect sense, see the commit I'll post in a moment.